docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceFactoryInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Drupal\foia_webform;
+
+use Drupal\node\NodeInterface;
+
+/**
+ * Provides an interface for submitting the request using the preferred method.
+ */
+interface FoiaSubmissionServiceFactoryInterface {
+
+  /**
+   * Determines which method to use for submitting the request.
```

Comment:
In addition to determining which method to use, it returns an instantiated service. Maybe we can change this to:
`Returns appropriate service for submitting the request based on component.`